### DISCUSSION TITLE
How to extract layers from psd file using ImageMagick

### DISCUSSION THREAD

**Discussion by SeenaThite** [ImageMagick/ImageMagick#5122] [Help]
Hi,

I am reading a psd file using ImageMagick. The psd file is a multi-layered. I want to extract each layer separately from this file and save it as bmp.
I am using ImageMagick 7 on Windows platform. Is there any C++ API of magick++ that can be used to extract layers from psd file?

**Comment by snibgo** [comment]
Perhaps by "layers", you mean images in the PSD file that IM can see. To test this:
```
magick yourfile.psd info:
```
Does that show your required layers? If it does, then...
```
magick yourfile.psd out.bmp
```
... will create one BMP file per image. You can write that as a CPP program if you want.
